zipline/finance/controls.py

```diff
@@ -75,6 +78,16 @@ def fail(self, asset, amount, datetime, metadata=None):
                                       datetime=datetime,
                                       constraint=constraint)
 
+    def log(self, asset, amount, datetime, metadata=None):
+        constraint = repr(self)
+        if metadata:
+            constraint = "{constraint} (Metadata: {metadata})".format(
```

Comment:
This is duplicated between `fail` and `log`.  We should probably refactor to something like `_format_error_message`.